baseframe/forms/widgets.py

```diff
@@ -36,6 +36,23 @@ def __call__(self, field, **kwargs):
         return HTMLString(''.join(html))
 
 
+class Select2Widget(Select):
+    """
+    Add support of choices with ``optgroup`` to the ``Select`` widget.
+    """
+    def __call__(self, field, **kwargs):
+        kwargs.setdefault('id', field.id)
+        kwargs.pop('type', field.type)
+        if field.multiple:
+            kwargs['multiple'] = 'multiple'
+        html = ['<select %s>' % html_params(name=field.name, **kwargs)]
```

Comment:
This should have a `class="select2"` so that we don't clobber other select inputs. Add it via kwargs just in case there is an existing class argument.